lib/item_widget.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:lists/item.dart';
+
+/// ItemWidget:
+///   - a widget representing a single item
+///   in a list (see Item)
+class ItemWidget extends StatefulWidget {
+  final Item item;
+  const ItemWidget(this.item, {super.key});
+
+  @override
+  State<ItemWidget> createState() => _ItemWidgetState();
+}
+
+class _ItemWidgetState extends State<ItemWidget> {
+  @override
+  Widget build(BuildContext context) {
+    final itemTextStyle = Theme.of(context).textTheme.titleLarge;
+
+    return ListTile(
+      title: TextFormField(
```

Comment:
I'm not sure this is the best design. It looks strange to have a list of `TextFormField` on the screen.   

This is the way I was envisioning it:  
* I would recommend replacing the `TextFormField` with `Text`
* Then add an `onTap` callback to the `ListTile`.  
* When the user taps a `ListTile`, open an `AlertDialog` that contains the `TextFormField`.
* The user enters the updated text into the dialog, and then presses some kind of confirmation button.
* When the confirmation button is tapped, the `Item` is updated with the new text.

Reply:
Adding a dialog does have the drawback that it adds an extra tap for the user, to confirm the dialog. Your current solution allows the user to jump from editing one item to another quickly, which is very nice. You could totally keep your current implementation, but you need to move the `TextEditingController` to `_ItemWidgetState`.  
* Use the `initialValue` parameter of `TextFormField` to set the initial value from `widget.item`.
* Use the `onChanged` parameter of `TextFormField` to get the updated value and store in `widget.item`.